### PR TITLE
Add path_delimiter to split with any char

### PR DESCRIPTION
**Pull request by ganmacs**
Signed-off-by: Yuta Iwama <ganmacs@gmail.com>

**Which issue(s) this PR fixes**: 
Fixes https://github.com/fluent/fluentd/pull/2001

**What this PR does / why we need it**: 

Add path_delimiter to split with any char
**Docs Changes**:

needed?

**Release Note**: 

same as the title

### CODE REVIEW

File: lib/fluent/plugin/in_tail.rb [hunk 1]
```diff
@@ -35,6 +35,8 @@ class TailInput < Fluent::Plugin::Input
 
     helpers :timer, :event_loop, :parser, :compat_parameters
 
+    RESERVED_CHARS = ['/', '*', '%'].freeze
```
Comment (cosmo0920) [inline lib/fluent/plugin/in_tail.rb]: These characters should be reserved characters for Unix-like systems.
MSDN says that the following characters are reserved characters in NTFS:

* `< (less than)`
* `> (greater than)`
* `: (colon)`
* `" (double quote)`
* `/ (forward slash)`
* `\ (backslash)`
* `| (vertical bar or pipe)`
* `? (question mark)`
* `* (asterisk)`


Should we also consider them here?

ref: https://docs.microsoft.com/en-us/windows/win32/fileio/naming-a-file#naming-conventions

Comment (ganmacs) [reply lib/fluent/plugin/in_tail.rb]: `RESERVED_CHARS` is fluentd's reserved chars, not UNIX-like system's or MSDN's one. 
I think it's good that users can use as many chars as possible on their own responsibility. 
WDYT?

Comment (cosmo0920) [reply lib/fluent/plugin/in_tail.rb]: I'm thinking of these reserved characters handling....
At least, I want to handle `\ (backslash)` as reserved character on Windows, if we can.

Comment (cosmo0920) [reply lib/fluent/plugin/in_tail.rb]: > `RESERVED_CHARS` is fluentd's reserved chars, not UNIX-like system's or MSDN's one.
> I think it's good that users can use as many chars as possible on their own responsibility.
> WDYT?

 I see. It's reasonable for me. :+1:

**Pull request merged**